app/models/concerns/alertable.rb

```diff
@@ -40,18 +49,36 @@ def remove_alert_on_save
   end
 
   def remove_field_change_alerts
-    alerts_on_change.each { |_, form_name| remove_alert(form_name) }
+    alerts_on_change.each do |_, conf_record|
+      next if conf_record.alert_strategy == AlertStrategy::ASSOCIATED_USERS
+
+      remove_alert(conf_record.form_section_unique_id)
+    end
   end
 
   def add_alert_on_field_change
-    return unless owned_by != last_updated_by
     return unless alerts_on_change.present?
 
     changed_field_names = changes_to_save_for_record.keys
-    alerts_on_change.each do |field_name, form_name|
+    alerts_on_change.each do |field_name, conf_record|
       next unless changed_field_names.include?(field_name)
 
-      add_alert(alert_for: FIELD_CHANGE, date: Date.today, type: form_name, form_sidebar_id: form_name)
+      add_field_alert(conf_record)
+    end
+  end
+
+  def add_field_alert(conf_record)
+    case conf_record.alert_strategy
```

Comment:
make sure you run rubocop for all the files added/modified
please leave a comment that this is partially implemented.